ambry-store/src/test/java/com.github.ambry.store/StoreFindTokenTest.java

```diff
@@ -226,9 +217,22 @@ private void doSerDeTest(StoreFindToken token, Short... versions) throws IOExcep
     for (Short version : versions) {
       DataInputStream stream = getSerializedStream(token, version);
       StoreFindToken deSerToken = StoreFindToken.fromBytes(stream, STORE_KEY_FACTORY);
+      assertEquals("Stream should have ended ", 0, stream.available());
+      assertEquals("Version mismatch for token ", version.shortValue(), deSerToken.getVersion());
       compareTokens(token, deSerToken);
       assertEquals("SessionId does not match", token.getSessionId(), deSerToken.getSessionId());
+      if (version == StoreFindToken.VERSION_2) {
+        assertEquals("IncarnationId mismatch ", token.getIncarnationId(), deSerToken.getIncarnationId());
+      }
+      stream = new DataInputStream(new ByteBufferInputStream(ByteBuffer.wrap(deSerToken.toBytes())));
```

Comment:
Sorry. I tried to think through why I added it, but couldn't find a reason. Will remove it for now. If you can think of a reason why it could fail, do let me know.